src/level.d.ts

```diff
@@ -0,0 +1,168 @@
+// readLevel, writeLevel
+
+// 1.8.8
+export interface LevelDatFull {
+    WorldGenSettings?: WorldGenSettings;
+    RandomSeed:           number[];
+    generatorName?:        string;
+    BorderCenterX:        number;
+    BorderCenterZ:        number;
+    Difficulty:           number;
+    DifficultyLocked:     number;
+    BorderSizeLerpTime:   number[];
+    Version?: {
+      Name: string
+      // id, snapshot
+    }
+    /** 0,1 */
+    raining:              number;
+    Time:                 number[];
+    GameType:             number;
+    MapFeatures:          number;
+    BorderDamagePerBlock: number;
+    BorderWarningBlocks:  number;
+    BorderSizeLerpTarget: number;
+    DayTime:              number[];
+    initialized:          number;
+    allowCommands:        number;
+    SizeOnDisk:           number[];
+    GameRules:            GameRules;
+    Player:               Player;
+    SpawnY:               number;
+    rainTime:             number;
+    thunderTime:          number;
+    SpawnZ:               number;
+    hardcore:             number;
+    SpawnX:               number;
+    clearWeatherTime:     number;
+    thundering:           number;
+    generatorVersion?:     number;
+    version:              number;
+    BorderSafeZone:       number;
+    generatorOptions?:     string;
+    LastPlayed:           number[];
+    BorderWarningTime:    number;
+    LevelName:            string;
+    BorderSize:           number;
+}
+
+export interface GameRules {
+  doTileDrops:         string;
+  doFireTick:          string;
+  reducedDebugInfo:    string;
+  naturalRegeneration: string;
+  doMobLoot:           string;
+  keepInventory:       string;
+  doEntityDrops:       string;
+  mobGriefing:         string;
+  randomTickSpeed:     string;
+  commandBlockOutput:  string;
+  doMobSpawning:       string;
+  logAdminCommands:    string;
+  sendCommandFeedback: string;
+  doDaylightCycle:     string;
+  showDeathMessages:   string;
+}
+
+export interface Player {
```

Comment:
Do we have these typings anywhere else? IMO it would be extremely useful to share them them across projects (note you can use it like this: `/** @type {import('prismarine-provider-anvil/src/types').Player} */`

Reply:
these types are specific to this module